### PR TITLE
copr: support load_based_replica_read_threshold

### LINKED ISSUE #41664: Support load-based replica read
## Enhancement

See https://github.com/tikv/rfcs/pull/105 for the TiKV design doc.

To support this feature in TiDB, add a session and global sysvar `tidb_load_based_replica_read_threshold`.

If the estimated waiting duration at TiKV exceeds the threshold, replica read is automatically enabled.

### PR DESCRIPTION
**Pull request by sticnarf**
Signed-off-by: Yilin Chen <sticnarf@gmail.com>

### What problem does this PR solve?

Issue Number: close #41664

Refers to the TiKV RFC https://github.com/tikv/rfcs/pull/105 and support setting threshold for coprocessor requests.

### What is changed and how it works?

Add a session and global sysvar `tidb_load_based_replica_read_threshold`.

If the estimated waiting duration at TiKV exceeds the threshold, replica read is automatically enabled.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by cfzjywxk — APPROVED** [review]

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -1137,6 +1137,7 @@ func (worker *copIteratorWorker) handleTaskOnce(bo *Backoffer, task *copTask, ch
 		TaskId:            worker.req.TaskID,
 		RequestSource:     task.requestSource.GetRequestSource(),
 		ResourceGroupName: worker.req.ResourceGroupName,
+		BusyThresholdMs:   uint32(worker.req.StoreBusyThreshold.Milliseconds()),
```
Comment (you06) [inline store/copr/coprocessor.go:1140]: I'm worried about how to deal with the batched tasks, if we send them to the replica of the main task, there will be many region miss errors...

**Review by you06 — APPROVED** [review]

**Comment by you06** [comment]
@cfzjywxk I think we can merge this PR.

**Pull request merged**